cmd/prometheus/main.go

```diff
@@ -94,13 +97,8 @@ func main() {
 
 	a.HelpFlag.Short('h')
 
-	a.Flag("log.level",
-		"Only log messages with the given severity or above. One of: [debug, info, warn, error, fatal]").
-		Default("info").StringVar(&cfg.logLevel)
-
-	a.Flag("log.format",
-		`Set the log target and format. Example: "logger:syslog?appname=bob&local=7" or "logger:stdout?json=true"`).
-		Default("logger:stderr").StringVar(&cfg.logFormat)
+	a.Flag(promlog.LevelFlagName, promlog.LevelFlagHelp).
```

Comment:
> I think the right solution is to write a skeleton generator that creates the basic Go files. That would be generally quite useful for people to quickly write their own exporters.

I think that'd be useful in general, and it's already somewhere in my backlog of ideas.

> We can than add new flags etc. to the generated files. Then the generator can just be re-run against a repo and one can easily verify potential breaking changes by running git diff and remove them if necessary.
> There should be generally no need to modify the generated structures significantly so no hard-to-resolve diffs should occur.

This has been done countless times before. From everything I've seen and read, the second you modify the generated files this breaks down. For example this works for proto and lex files, but not for Ruby on Rails or maven projects.

I think a library function is the way to go here. We can recommend/generate boilerplate that does the plumbing with that function, but the boilerplate itself should very rarely change.